javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentClassLoader.java

```diff
@@ -158,7 +167,14 @@ public Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundExce
       }
       // search from parent and urls added to this loader
       if (clazz == null) {
-        clazz = super.loadClass(name, false);
+        try {
+          clazz = Class.forName(name, false, this.getParent());
+        } catch (ClassNotFoundException e) {
+          // ignore
+        }
+      }
+      if (clazz == null) {
+        clazz = super.findClass(name);
```

Comment:
After changing the `InstrumentationModuleClassLoader.defineClassWithPackage` this probably isn't necessary any more. Either revert or add a comment that this is here to avoid calling `super.loadClass`.

Reply:
I've also removed the `loadClass` override in `AgentClassLoader`. Remove it in `InstrumentationModuleClassLoader` is not possible, because then the `LoadClassAdvice` itself becomes recursive (not the bootstrapping this time):

```
[otel.javaagent 2024-10-24 12:42:55:462 +0200] [main] DEBUG io.opentelemetry.javaagent.tooling.instrumentation.indy.IndyBootstrap - Fixing nested instrumentation invokedynamic instruction bootstrapping for instrumented class java.lang.ClassLoader and advice io.opentelemetry.javaagent.instrumentation.internal.classloader.LoadInjectedClassInstrumentation$LoadClassAdvice.onEnter, the instrumentaiton should be active now
[otel.javaagent 2024-10-24 12:42:55:463 +0200] [otel-javaagent-transform-safe-logger] DEBUG io.opentelemetry.javaagent.tooling.AgentInstaller$TransformLoggingListener - Failed to handle io.opentelemetry.javaagent.ClassLoaderData$$29539e36 for transformation on class loader io.opentelemetry.javaagent.tooling.util.ClassLoaderMap$1@29539e36
java.lang.StackOverflowError
	at io.opentelemetry.javaagent.instrumentation.internal.classloader.LoadInjectedClassInstrumentation$LoadClassAdvice.onEnter(LoadInjectedClassInstrumentation.java:59)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	at io.opentelemetry.javaagent.instrumentation.internal.classloader.LoadInjectedClassInstrumentation$LoadClassAdvice.onEnter(LoadInjectedClassInstrumentation.java:59)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	at io.opentelemetry.javaagent.instrumentation.internal.classloader.LoadInjectedClassInstrumentation$LoadClassAdvice.onEnter(LoadInjectedClassInstrumentation.java:59)
	...
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	at io.opentelemetry.javaagent.instrumentation.internal.classloader.LoadInjectedClassInstrumentation$LoadClassAdvice.onEnter(LoadInjectedClassInstrumentation.java:59)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	at io.opentelemetry.javaagent.instrumentation.internal.classloader.LoadInjectedClassInstrumentation$LoadClassAdvice.onEnter(LoadInjectedClassInstrumentation.java:59)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
	at io.opentelemetry.javaagent.tooling.ignore.IgnoredClassLoadersMatcher.loadsExpectedClass(IgnoredClassLoadersMatcher.java:78)
	at io.opentelemetry.javaagent.tooling.ignore.IgnoredClassLoadersMatcher.delegatesToBootstrap(IgnoredClassLoadersMatcher.java:69)
```

If you prefer to, we could also use `CallDepth` in the `LoadClassAdvice` instead I think.

Reply:
I think it is fine as it is for now.